content/docs/reference/ecs-partial-task-definitions.md

```diff
@@ -0,0 +1,189 @@
+---
+title: "ECS Partial Task Definitions"
+description: "Our way of managing ECS Services using Terraform and GitHub Actions."
+slug: "ecs-partial-task-definitions"
+tags:
+- "ECS"
+- "Terraform"
+- "GitHub Actions"
+categories:
+- "Community Resources"
+---
+
+
+# ECS Partial Task Definitions
+
+This document describes what partial task definitions are and how we can use them to set up ECS services using Terraform and GitHub Actions.
+
+## The Problem
+
+Managing ECS Services is challenging. Ideally, we want our services to be managed by terraform so everything is living in code. However, we also want to update the task definition via GitOps as through the GitHub release lifecycle. This is challenging because terraform can create the task definition, but if updated by the application repository, the task definition will be out of sync with the terraform state.
+
+Managing it entirely through terraform means we cannot easily update the newly built image by the application repository unless we directly commit to the infrastructure repository, which is not ideal.
+
+Managing it entirely through the application repository means we cannot codify the infrastructure and have to hardcode ARNs, secrets, and other infrastructure-specific configurations.
+
+## Introduction
+
+ECS Partial task definitions is the idea of breaking the task definition into smaller parts. This allows for easier management of the task definition and makes it easier to update the task definition.
+
+We do this by setting up terraform to manage a portion of the task definition, and the application repository to manage another portion.
+
+The terraform (infrastructure) portion is created first, it will create an ECS Service in ECS then upload the task definition json to S3 as `task-template.json`.
```

Comment:
```suggestion
The terraform (infrastructure) portion is created first. It will create an ECS Service in ECS then upload the task definition json to S3 as `task-template.json`.
```